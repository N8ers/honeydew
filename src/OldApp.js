import PropTypes from "prop-types"

function OldApp(props) {
  const { message } = props
  return (
    <div>
      <header>
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
          Learn React
        </a>
      </header>

      <div>
        <h2>Tsuki Baby Cat</h2>
        <p data-testid="pTag">He is just this big</p>
      </div>

      <div>
        <p data-testid="propMessage">{message}</p>
      </div>
    </div>
  )
}

OldApp.propTypes = {
  message: PropTypes.string,
}

OldApp.defaultProps = {
  message: "default message",
}

export default OldApp
