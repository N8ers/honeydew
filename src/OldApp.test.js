import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"
import OldApp from "./OldApp"

test("renders learn react link", () => {
  render(<OldApp />)
  const linkElement = screen.getByText(/learn react/i)
  expect(linkElement).toBeInTheDocument()
})

test("display p tag", () => {
  render(<OldApp message="hi there!" />)

  const pTag = screen.getByTestId("pTag")
  expect(pTag).toContainHTML('<p data-testid="pTag">He is just this big</p>')
  expect(screen.getByText("He is just this big")).toBeInTheDocument()

  expect(pTag).toMatchSnapshot()
})

test("renders default props", () => {
  render(<OldApp />)

  const messageProp = screen.getByTestId("propMessage")
  expect(messageProp).toContainHTML(
    '<p data-testid="propMessage">default message</p>'
  )

  expect(messageProp).toMatchSnapshot()
})

test("renders custom prop", () => {
  render(<OldApp message="hi there!" />)

  const messageProp = screen.getByTestId("propMessage")
  expect(messageProp).toContainHTML(
    '<p data-testid="propMessage">hi there!</p>'
  )
  expect(messageProp).toMatchSnapshot()
})
