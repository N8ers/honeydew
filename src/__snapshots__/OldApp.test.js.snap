// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display p tag 1`] = `
<p
  data-testid="pTag"
>
  He is just this big
</p>
`;

exports[`renders custom prop 1`] = `
<p
  data-testid="propMessage"
>
  hi there!
</p>
`;

exports[`renders default props 1`] = `
<p
  data-testid="propMessage"
>
  default message
</p>
`;
